crates/services/txpool_v2/src/pool_worker.rs

```diff
@@ -291,20 +296,23 @@ where
                     None => return true,
                 }
             }
-            _ = self.request_remove_receiver.recv_many(&mut remove_buffer, MAX_PENDING_REMOVE_POOL_REQUESTS) => {
-                if remove_buffer.is_empty() {
+            _ = self.request_update_tx_receiver.recv_many(&mut update_tx_buffer, MAX_PENDING_REMOVE_POOL_REQUESTS) => {
+                if update_tx_buffer.is_empty() {
                     return true;
                 }
-                for remove in remove_buffer {
-                    match remove {
-                        PoolRemoveRequest::ProcessBlock { block_result } => {
+                for update_tx in update_tx_buffer {
+                    match update_tx {
+                        PoolTxUpdateRequest::ProcessBlock { block_result } => {
                             self.process_block(block_result);
                         }
-                        PoolRemoveRequest::SkippedTransactions { dependents_ids } => {
+                        PoolTxUpdateRequest::SkippedTransactions { dependents_ids } => {
                             self.remove_skipped_transactions(dependents_ids);
                         }
-                        PoolRemoveRequest::TxAndCoinDependents { tx_and_dependents_ids } => {
-                            self.remove_and_coin_dependents(tx_and_dependents_ids);
+                        PoolTxUpdateRequest::ExpiredTransactionsAndDependents { tx_and_dependents_ids } => {
+                            self.expired_transactions_and_dependents(tx_and_dependents_ids);
+                        }
+                        PoolTxUpdateRequest::PreconfirmedTransactions { txs } => {
+                            self.preconfirmed_transactions(txs);
```

Comment:
can we pass an iterator instead?
```suggestion
                            self.preconfirmed_transactions(txs.iter());
```